### PR TITLE
query and update animals table

### PR DESCRIPTION
**Pull request by Ellyboi**
**Below requirement implemented:**

- Add a column species of type string to your animals table. Modify your schema.sql file.
- Inserted the given data
- set the species column to unspecified and verified that change. 
- Inside a transaction
       - Updated the animals table by setting the species column to digimon for all animals that have a name ending in mon.
       - Updated the animals table by setting the species column to pokemon for all animals that don't have species already set.
       - Verified the changes.
       - Commit the transaction.
       - Verify that changes persist after commit.
- Wrote the queries

 **The screenshot below:**

ScreenshotAA
ScreenshotAB
ScreenshotAC
ScreenshotAD
ScreenshotAE
ScreenshotAF
ScreenshotAG
ScreenshotAH
ScreenshotAI
ScreenshotAJ
ScreenshotAK
ScreenshotAL
ScreenshotAM
ScreenshotAN

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -10,4 +10,15 @@ CREATE TABLE animals (
...
+
+-- Creating the animals table
+CREATE TABLE animals (
+    id INT,
+    name VARCHAR(100),
+    date_of_birth DATE,
+    escape_attempts INT,
+    neutered BOOLEAN,
+    weight_kg DECIMAL(10,2),
+    species VARCHAR(100)
```
Comment (topeogunleye) [inline schema.sql]: Hi, thank you for your submission. I have a few comments on your code:

The assignment asked you to use the ALTER TABLE command to add a new column called species with a string type. However, you Created a new TABLE, then added the species column directly to the new CREATE TABLE statement. This does not demonstrate your understanding of how to modify an existing table using SQL commands. Please revise your code to use the ALTER TABLE command as instructed. Please kindly find below an example:

`ALTER TABLE animals ADD species varchar(255);`

File: data.sql [hunk 2]
```diff
@@ -7,3 +7,13 @@ VALUES
...
+-- Inserting additional data into the animals table
+INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg, species)
+VALUES
+    (5, 'Charmander', '2020-02-08', 0, false, -11, NULL),
+    (6, 'Plantmon', '2021-11-15', 2, true, -5.7, NULL),
+    (7, 'Squirtle', '1993-04-02', 3, false, -12.13, NULL),
+    (8, 'Angemon', '2005-06-12', 1, true, -45, NULL),
+    (9, 'Boarmon', '2005-06-07', 7, true, 20.4, NULL),
+    (10, 'Blossom', '1998-10-13', 3, true, 17, NULL),
+    (11, 'Ditto', '2022-05-14', 4, true, 22, NULL);
```
Comment (topeogunleye) [inline data.sql]: Hi, thank you for your submission. I have one required comment on your code:

- **Required**: You added a species column to the INSERT INTO statements, which was not part of the instructions. The instructions only asked for name, date_of_birth, escape_attempts, neutered, and weight_kg columns. Please remove the species column from your code and resubmit.


Please make the required change and resubmit your code for another review. If you have any questions or need any help, feel free to reach out to me. Thank you for your hard work and cooperation.

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @Ellyboi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

> ## (Highlights) Good Points: 💪 
- The student used various SQL functions and clauses to answer the questions about the animals table. The student used the COUNT(*) function to count the total number of animals and the number of animals that have never tried to escape. The student used the AVG function to calculate the average weight of animals and the average number of escape attempts per animal type of those born between 1990 and 2000. The student used the MAX and MIN functions to find the maximum escape attempts for neutered and not neutered animals, and the minimum and maximum weight for each type of animal. The student used the GROUP BY clause to group the results by neutered status and species. The student used the WHERE clause and the BETWEEN operator to filter the results by date of birth. ✅ 
- Used a transaction to update the animals table by setting the species column to unspecified, and then rolled back the change. The student also verified the change and the rollback by selecting the species column from the animals table. This showed the student’s understanding of how to use transactions and rollbacks in SQL ✅ 

## Required Changes ♻️
_Check the comments under the review._
- I kindly suggest that you add instructions for cloning your project locally. 
For Example:
```
  cd vet-clinic
  git clone git@github.com/Ellyboi/vet-clinic-database.git
```
image

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- For future pull requests, please notice that your branch name `query_update` does not follow the convention of being named after a feature or action. For example, you could name your branch `added-query-update`. This makes it easier to track the changes and the purpose of each branch.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql [hunk 3]
```diff
@@ -8,3 +8,95 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name <> 'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+
+ALTER TABLE animals ADD COLUMN species VARCHAR;
+-- Starting the transaction
```
Comment (TadesseDev) [inline queries.sql]: - [ ] Dear @Ellyboi, Good job using the `ALTER TABLE` statement to add the column as requested. But while doing so I notice you have added this into your `queries.sql` file. Please remember everything that creates or changes the table structure should be saved in the schema.sql file. Please consider moving this statement to the `schema.sql` file.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @Ellyboi 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

199733072-bdacca19-8df9-43a7-be93-3aff2f1c15c8

 ## Highlights
 - [x] Good job making most of the changes requested by the previous reviewer. 🙏


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- N/A



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bobb-Rob — APPROVED** [review]
## STATUS: APPROVED ✔️ 🍾 🍾 🍾 🥳 🥳 🥳 

Hi @Ellyboi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ No Linter error
✅ Correct use of GitHub flow
✅ Very descriptive Pull Request(PR)

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**